src/Microsoft.TemplateEngine.Core/Operations/InlineMarkupConditional.cs

```diff
@@ -0,0 +1,233 @@
+
+        public bool WholeLine { get; }
+
+        public IOperation GetOperation(Encoding encoding, IProcessorState processorState)
+        {
+            TokenTrie structureTrie = new TokenTrie();
+            TokenTrie closeConditionTrie = new TokenTrie();
+            TokenTrie scanBackTrie = new TokenTrie();
+
+            byte[] openOpenElementTokenBytes = processorState.Encoding.GetBytes(Tokens.OpenOpenElementToken);
+            scanBackTrie.AddToken(openOpenElementTokenBytes);
+            int openOpenElementToken = structureTrie.AddToken(openOpenElementTokenBytes);
+            int openCloseElementToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.OpenCloseElementToken));
+            int closeCloseElementToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.CloseElementTagToken));
+
+            int selfClosingElementEndToken = -1;
+            if (Tokens.SelfClosingElementEndToken != null)
+            {
+                selfClosingElementEndToken = structureTrie.AddToken(processorState.Encoding.GetBytes(Tokens.SelfClosingElementEndToken));
+            }
+
+            closeConditionTrie.AddToken(processorState.Encoding.GetBytes(Tokens.CloseConditionExpression));
+            MarkupTokenMapping mapping = new MarkupTokenMapping(
+                openOpenElementToken,
+                openCloseElementToken,
+                closeCloseElementToken,
+                selfClosingElementEndToken
+            );
+
+            IReadOnlyList<byte[]> start = new[] { processorState.Encoding.GetBytes(Tokens.OpenConditionExpression) };
+            return new Impl(this, start, structureTrie, closeConditionTrie, scanBackTrie, mapping, _id);
+        }
+
+        public class Impl : IOperation
+        {
+            private readonly TokenTrie _closeConditionTrie;
+            private readonly InlineMarkupConditional _definition;
+            private readonly MarkupTokenMapping _mapping;
+            private readonly TokenTrie _scanBackTrie;
+            private readonly TokenTrie _structureTrie;
+
+            public Impl(InlineMarkupConditional definition, IReadOnlyList<byte[]> tokens, TokenTrie structureTrie, TokenTrie closeConditionTrie, TokenTrie scanBackTrie, MarkupTokenMapping mapping, string id)
+            {
+                _definition = definition;
+                Id = id;
+                Tokens = tokens;
+                _mapping = mapping;
+                _structureTrie = structureTrie;
+                _scanBackTrie = scanBackTrie;
+                _closeConditionTrie = closeConditionTrie;
+            }
+
+            public string Id { get; }
+
+            public IReadOnlyList<byte[]> Tokens { get; }
+
+            public int HandleMatch(IProcessorState processor, int bufferLength, ref int currentBufferPosition, int token, Stream target)
+            {
+                bool flag;
+                if (processor.Config.Flags.TryGetValue("conditionals", out flag) && !flag)
```

Comment:
This is by design, as it's the same family of operations as regular conditionals - but I agree it's a bit odd. The reason for internalizing the global flag checks is for the flags operator to determine whether it's got the turn flags on flag (when flags processing is currently off)